services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -477,32 +528,213 @@ async def trigger_connected_service_retrieve(
+
+    # let's now check if anyone has the project in use somehow
     with managed_resource(user_id, None, app) as rt:
-        # checks who is using it
-        users_of_project = await rt.find_users_of_resource("project_id", project_uuid)
-        usernames = [await get_user_name(app, uid) for uid in set(users_of_project)]
-        assert (
-            len(usernames) <= 1
-        )  # nosec  # currently not possible to have more than 1
-
-        # based on usage, sets an state
-        is_locked: bool = len(usernames) > 0
-        if is_locked:
+        user_session_id_list: List[Tuple[int, str]] = await rt.find_users_of_resource(
+            PROJECT_ID_KEY, project_uuid
+        )
+    set_user_ids = {x for x, _ in user_session_id_list}
+
+    assert (
+        len(set_user_ids) <= 1
+    )  # nosec  # NOTE: A project can only be opened by one user in one tab at the moment
+
+    if not set_user_ids:
+        # no one has the project, so it is unlocked and closed.
+        log.debug("project [%s] is not in use", project_uuid)
+        return ProjectLocked(value=False, status=ProjectStatus.CLOSED)
+
+    log.debug(
+        "project [%s] might be used by the following users: [%s]",
+        project_uuid,
+        set_user_ids,
+    )
+    usernames: List[Dict[str, str]] = [
+        await get_user_name(app, uid) for uid in set_user_ids
+    ]
+    # let's check if the project is opened by the same user, maybe already opened or closed in a orphaned session
+    if set_user_ids.issubset({user_id}):
+        if not await _user_has_another_client_open(user_session_id_list, app):
+            # in this case the project is re-openable by the same user until it gets closed
+            log.debug(
+                "project [%s] is in use by the same user [%s] that is currently disconnected, so it is unlocked for this specific user and opened",
+                project_uuid,
+                set_user_ids,
+            )
             return ProjectLocked(
-                value=is_locked,
-                owner=Owner(user_id=users_of_project[0], **usernames[0]),
+                value=False,
+                owner=Owner(user_id=list(set_user_ids)[0], **usernames[0]),
+                status=ProjectStatus.OPENED,
             )
-        return ProjectLocked(value=is_locked)
+    # the project is opened in another tab or browser, or by another user, both case resolves to the project being locked, and opened
+    log.debug(
+        "project [%s] is in use by another user [%s], so it is locked",
+        project_uuid,
+        set_user_ids,
+    )
+    return ProjectLocked(
+        value=True,
+        owner=Owner(user_id=list(set_user_ids)[0], **usernames[0]),
+        status=ProjectStatus.OPENED,
+    )
+
+
+async def get_project_states_for_user(
+    user_id: int, project_uuid: str, app: web.Application
+) -> ProjectState:
+    # for templates: the project is never locked and never opened. also the running state is always unknown
```

Comment:
is unknown always used for templates? If yes, maybe a better name?

Reply:
no. this is a running state. what is the running state of templates? they never run. if you prefer I could put not_started.